### PR TITLE
Integration specs/n+1 problems

### PR DESCRIPTION
**Pull request by jlberbesi**
Hello fellow code reviewer, In this opportunity we did the following tasks:

In this pull request we made the following changes:
Make sure that the N+1 problem is solved when fetching all posts and their comments for a user

Integration specs
 Use Capybara to write integration tests for each view in your project.

 Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.
 
User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page:

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page:

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.

Before User index:

imagen

After User Index

imagen

Before Users Shows:

imagen

After Users Shows:

imagen

Before post index:

imagen

After post index:

imagen

All tests:

imagen

### CODE REVIEW

File: spec/features/users_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  before do
+    @user1 = User.create!(name: 'User 1', email: 'user1@example.com')
+    visit users_path
+  end
+
+  it 'redirects to user show page on click' do
+    click_on @user1.name
+    expect(page).to have_current_path(user_path(@user1))
+  end
+end
```
Comment (Olamarx) [inline spec/features/users_index_spec.rb]: - [ ] I can see you have only one test here whereas the project milestone requires 4 tests from this file, kindly add the remaining tests.
image

File: spec/features/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  it 'shows the user details and their posts' do
+    expect(page).to have_content(@user.name)
+    expect(page).to have_content("Number of posts: #{@user.posts.count}")
+  end
+
+  it 'redirects to the full post list on click' do
+    click_on('See all posts', match: :first)
+    expect(page).to have_current_path(user_path(@user))
+  end
```
Comment (Olamarx) [inline spec/features/user_show_spec.rb]: - [ ] Just as I have pointed previously, this is also required to have a total test of 8 but you have only two test written, kindly add the rest of the test.
image

File: spec/features/posts_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    @post = @user.posts.create!(title: 'Sample Post', text: 'This is a sample post')
+  end
+
+  it 'shows details of a post' do
+    visit post_path(@post)
+    expect(page).to have_content(@post.title)
+    expect(page).to have_content(@post.text)
+    expect(page).to have_content('Comments: ')
+    expect(page).to have_content('Likes: ')
+  end
```
Comment (Olamarx) [inline spec/features/posts_show_spec.rb]: - [ ] Just as others I have pointed, this test is to total 7 but I can see you have a chained test, I would encourage you to write the tests separately and write all the tests required for this file.
image

File: spec/features/posts_index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      expect(page).to have_content(post.title)
+      expect(page).to have_content(post.text[0..20])
+    end
+  end
+
+  it 'redirects to the post show page on click' do
+    visit user_posts_path(@user)
+    click_on @user.posts.first.title
+    expect(current_path).to eq(post_path(@user.posts.first))
+  end
```
Comment (Olamarx) [inline spec/features/posts_index_spec.rb]: - [ ] Here, you only have two tests written whereas, it is expected to be 10. Kindly add the rest 
image

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @jlberbesi, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Some tests written ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/users_index_spec.rb (hunk 1 above)
Comment (jlberbesi) [reply spec/features/users_index_spec.rb]: imagen

File: spec/features/user_show_spec.rb (hunk 2 above)
Comment (jlberbesi) [reply spec/features/user_show_spec.rb]: imagen

File: spec/features/posts_show_spec.rb (hunk 3 above)
Comment (jlberbesi) [reply spec/features/posts_show_spec.rb]: imagen

File: spec/features/posts_index_spec.rb (hunk 4 above)
Comment (jlberbesi) [reply spec/features/posts_index_spec.rb]: imagen

**Review by Davidosky007 — APPROVED** [review]
Hi @jlberbesi 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* Well-written PR 👍 
* No linter errors 👍 
* All tests passed 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**